docs/ballerina-to-oas/proposals/hateoas_support.md

```diff
@@ -0,0 +1,164 @@
+# Proposal: Map Ballerina HATEOAS Links to OpenAPI Specification
+
+_Owners_: @SachinAkash01 
+_Reviewers_: @lnash94 @TharmiganK   
+_Created_: 2023/10/25  
+_Updated_: 2023/10/31  
+_Issue_: [#4788](https://github.com/ballerina-platform/ballerina-library/issues/4788)
+
+## Summary
+This proposal is to introduce the capabilities to support HATEOAS (Hypermedia As The Engine Of Application State) links 
+in the generated OpenAPI specification of the Ballerina OpenAPI tool. Note that in the rest of the proposal HATEOAS is 
+referred to as Hypermedia constraint.
+
+## Goals
+- Enhance the Ballerina OpenAPI tool by adding support for Hypermedia constraints in the generated OpenAPI specification.
+
+## Motivation
+OpenAPI is a widely used standard for documenting APIs. It allows developers to specify the structure and data types 
+expected in API requests and responses.
+
+Hypermedia constraints are one of the key principles in REST though it is often ignored. This principle emphasizes the 
+interconnection of resources, offering API users an experience similar to navigating the web. The existing tool for 
+generating OpenAPI specification from Ballerina code lacks support for including these Hypermedia constraints in the 
+OpenAPI specification. This means that when developers use Hypermedia constraints in their Ballerina code, this 
+information is not properly reflected in the OpenAPI specification.
+
+Overall, improving the OpenAPI tool to support Hypermedia constraints of the Ballerina programming language in the 
+generated OpenAPI specification is a worthwhile investment that will improve the overall developer experience.
+
+## Description
+There are two ways of generating links between the resources in Ballerina, manually defining links between resources and
+automatically generating links between resources.  The feature implementation focuses on integrating Ballerina 
+Hypermedia constraints into generated OpenAPI specifications to accurately document the API.  In this proposal we will 
+be addressing the automatic linking between the resources using `ResourceConfig` annotation in Ballerina.
+
+Currently in manual linking approach, we are generating the response component schema by including `*http:Links`. 
+There will be no changes to this behaviour of generating the OpenAPI specification.
+
+**Sample Ballerina Code (Manually defining links between resources):**
```

Comment:
We are not doing this part right ?

Reply:
Its the current way how the manual linking between resources is reflected in OAS I have mentioned that this current behavior will not be changed..